xfetus/improved-dsrgan/finetune_ddpm.py

```diff
@@ -0,0 +1,337 @@
+from os import path as osp
+
+from tqdm import tqdm
+import torch
+from torchvision import transforms
+from torch.utils.data import DataLoader
+import argparse
+import wandb
+from diffusers import DDIMScheduler, DDPMPipeline
+import torch.nn.functional as F
+import numpy as np
+import torch.nn as nn
+import matplotlib.pyplot as plt
+import copy
+from haar_pytorch import HaarForward, HaarInverse
+
+from loss import loss_img, loss_hf, loss_hfmse
+from ddpm_dataset import PrecomputedFetalPlaneDataset
+
+if __name__ == "__main__":
+
+    ##################
+    ##   1. SETUP   ##
+    ##################
+
+    # Command line aurgments - for script
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-d", "--dataset_path", help="File location of the fetal brain dataset", type=str)
+    parser.add_argument("-w", "--wandb_enabled", help="Enable weights and bias logging", type=bool)
+    parser.add_argument("-c", "--ddpm_checkpoint_path", default=None,  help="File location for the pre-trained DDPM model", type=str)
+    parser.add_argument("-l", "--ddpm_pa_loss", default=None,  help="Use DDPM-PA loss (as opposed to standard DDPM loss)", type=bool)

```

Comment:
DDPM-PA Loss is probably no longer needed.  We can keep in the code for now since it is an optional argument, but if we refactor, DDPM-PA should be removed.